x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowTask.java

```diff
@@ -149,13 +169,14 @@ public boolean equals(Object o) {
                 Objects.equals(leaderShardId, that.leaderShardId) &&
                 maxChunkSize == that.maxChunkSize &&
                 numConcurrentChunks == that.numConcurrentChunks &&
-                processorMaxTranslogBytes == that.processorMaxTranslogBytes;
+                processorMaxTranslogBytes == that.processorMaxTranslogBytes &&
+                Objects.equals(headers, that.headers);
```

Comment:
do you have equals/hashcode unit tests? If not can you add some and ensure the headers aspect is appropriately tested?

Reply:
This test: https://github.com/martijnvg/elasticsearch/blob/75ea05434003c2535c3b7e3e796dbd292fda1062/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/ShardFollowTaskTests.java#L17

also tests equals and hashcode, because it indirectly extends from `AbstractWireTestCase`.